lib/strava/models/mixins/upload_status.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Strava
+  module Models
+    module Mixins
+      module UploadStatus
+        extend ActiveSupport::Concern
+
+        included do
+          property 'id'
+          property 'external_id'
+          property 'error'
+          property 'status'
+          property 'activity_id'
+        end
+
+        def processing?
+          status.match?(/still being processed/i)
```

Comment:
Is there a more reliable way by checking `id`?

Reply:
sadly, I'd say no.

the `id` of the upload won't change, `activity_id` is only set when successful. the message in `status` seems to be the what's Strava chose to tell what's going.

Reply:
😞 thx

Reply:
We should still check whether `activity_id` is set, in which case it's definitely not processing.

Reply:
it's either processed and created an activity or failed 👍

Reply:
done